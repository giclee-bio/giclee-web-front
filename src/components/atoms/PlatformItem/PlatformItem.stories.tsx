import { Meta, StoryObj } from '@storybook/react'

import PlatformItem from './PlatformItem'

const meta: Meta<typeof PlatformItem> = {}

export default meta
type Story = StoryObj<typeof PlatformItem>

export const Index: Story = {
  args: {
    alt: 'nacal',
    caption: 'nacal',
    src: '/nacal.png',
  },
}
