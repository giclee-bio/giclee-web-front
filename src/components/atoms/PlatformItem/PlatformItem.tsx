import Image from 'next/image'

export type PlatformItems = {
  alt: string
  caption: string
  src: string
}
interface Props extends PlatformItems {
  handleSelectItem: (item: PlatformItems) => void
}

const PlatformItem: React.FC<Props> = ({ alt, caption, handleSelectItem, src }) => {
  return (
    <button className='w-20' onClick={() => handleSelectItem({ alt, caption, src })}>
      <div className='flex h-20 w-20 items-center justify-center rounded-xl bg-stone-100'>
        <Image alt={alt} className='rounded' height='48' src={src} width='48' />
      </div>
      <p className='mt-1 text-center text-sm'>{caption}</p>
    </button>
  )
}

export default PlatformItem
