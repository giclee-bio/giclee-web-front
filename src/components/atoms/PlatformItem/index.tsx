import PlatformItem from './PlatformItem'

export default PlatformItem
