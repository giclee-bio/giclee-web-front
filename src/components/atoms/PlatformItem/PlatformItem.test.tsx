import { render } from '@testing-library/react'

import PlatformItem from './PlatformItem'

describe('PlatformItem', () => {
  it('snapshot', async () => {
    const { asFragment } = render(<PlatformItem alt='nacal' caption='nacal' src='/nacal.png' />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
