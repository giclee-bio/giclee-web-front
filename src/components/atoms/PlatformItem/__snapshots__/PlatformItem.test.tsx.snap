// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformItem snapshot 1`] = `
<DocumentFragment>
  <button
    class="w-20"
  >
    <div
      class="flex h-20 w-20 items-center justify-center rounded-xl bg-stone-100"
    >
      <img
        alt="nacal"
        class="rounded"
        data-nimg="1"
        decoding="async"
        height="48"
        loading="lazy"
        src="/_next/image?url=%2Fnacal.png&w=96&q=75"
        srcset="/_next/image?url=%2Fnacal.png&w=48&q=75 1x, /_next/image?url=%2Fnacal.png&w=96&q=75 2x"
        style="color: transparent;"
        width="48"
      />
    </div>
    <p
      class="mt-1 text-center text-sm"
    >
      nacal
    </p>
  </button>
</DocumentFragment>
`;
